base/float.jl

```diff
@@ -414,7 +426,125 @@ muladd(x::T, y::T, z::T) where {T<:IEEEFloat} = muladd_float(x, y, z)
 # TODO: faster floating point fld?
 # TODO: faster floating point mod?
 
-rem(x::T, y::T) where {T<:IEEEFloat} = rem_float(x, y)
+function unbiased_exponent(x::T) where {T}
+    exp = (uintbits(x) & exponent_mask(T)) >> mantissa_width(T)
+    return exp
+end
+
+function explicit_mantissa_noinfnan(x::T) where {T}
+    subnormal = unbiased_exponent(x) > uinttype(T)(0) ? significand_mask(T) + uinttype(T)(1) : uinttype(T)(0)
+    return subnormal | (significand_mask(T) & uintbits(x))
+end
+
+function make_value(number::T, ep) where {T}
+    Tfloat = floattype(T)
+    Tint = inttype(T)
+    epint = unsafe_trunc(Tint,ep)
+    lz::inttype(T) = Core.Intrinsics.ctlz_int(number) - exponent_width(Tfloat)
```

Comment:
Any reason not to use `leading_zeros(reinterpret(Unsigned, number))` rather than calling into `Core.Intrinsics`?

Reply:
yeah. It's the difference between a UInt and an Int

Reply:
Specifically leading_zeros promotes to a 64 bit integer, while the intrinsic does not.